Src/IronPython/Runtime/MemoryView.cs

```diff
@@ -543,6 +552,95 @@ public sealed class MemoryView : ICodeFormattable, IWeakReferenceable {
             }
         }
 
+        public object this[[NotNull]PythonTuple index] {
+            get {
+                CheckBuffer();
+                return getAtFlatIndex(GetFlatIndex(index));
+            }
+            set {
+                CheckBuffer();
+                setAtFlatIndex(GetFlatIndex(index), value);
+            }
+        }
+
+        /// <summary>
+        /// Gets the "flat" index from the access of a tuple as if the
+        /// multidimensional tuple were layed out in contiguous memory.
+        /// </summary>
+        private int GetFlatIndex(PythonTuple tuple) {
+            int flatIndex = 0;
+            int tupleLength = tuple.Count;
+            int ndim = (int)this.ndim;
+            int firstOutOfRangeIndex = -1;
+
+            bool sawSlice = false;
+            bool sawInt = false;
+
+            // A few notes about the ordering of operations here:
+            // 1) CPython checks the types of the objects in the tuple
+            //    first, then the dimensions, then finally for the range.
+            //    Because we do a range check while we go through the tuple,
+            //    we have to remember that we had something out of range
+            // 2) CPython checks for a multislice tuple, then for all ints,
+            //    and throws an invalid slice key otherwise. We again try to
+            //    do this in one pass, so we remember whether we've seen an int
+            //    and whether we've seen a slice. If we haven't seen both, we
+            //    it means it's entirely ints or entirely slices
+            for (int i = 0; i < tupleLength; i++) {
+                object indexObject = tuple[i];
+                if (Converter.TryConvertToInt32(indexObject, out int indexValue)) {
```

Comment:
Not sure if it's worth it or if it makes it harder to understand, but you might be able able to replace `sawInt`/`sawTuple` with something like:
```C#
bool allInts = true;
for (int i = 0; i < tupleLength; i++) {
    object indexObject = tuple[i];
    if (allInts && Converter.TryConvertToInt32(...)) {
        // ...
    } else if (indexObject is Slice) {
        allInts = false;
    } else {
        throw PythonOps.TypeError("memoryview: invalid slice key");
    }
}

if (!allInts) {
    throw PythonOps.NotImplementedError("multi-dimensional slicing is not implemented");
}
```

Reply:
The only thing here is if we have a mix of ints and slices we need to throw the invalid slice key, and they wrote tests to explicitly check that

Reply:
Wait that does account for it

Reply:
I think it's the most clear with the two variables. With just `allInts` we need to catch the case in which we have an int, followed only by slices. It's possible to do it with only one variable like so:
```
else if (indexObject is Slice) {
    if (allInts && i > 0) {
        throw PythonOps.TypeError("memoryview: invalid slice key");
    }
    allInts = false;
}
```
but I don't think it provides a lot of benefit.